scripts/proof-gate.test.sh

```
@@ -0,0 +1,63 @@
+# exactly the class of bug this whole file exists to catch.
+grep -q 'REQUIRE_PROOF: ${{ inputs.require_proof }}' "$ROOT/action.yml" \
+  && report 0 'action.yml wires require_proof input to REQUIRE_PROOF' \
+  || report 1 'action.yml wires require_proof input to REQUIRE_PROOF'
+grep -q 'VCR_REQUIRE_PROOF: ${{ inputs.require_proof }}' "$ROOT/action.yml" \
+  && report 0 'action.yml wires require_proof input to VCR_REQUIRE_PROOF' \
+  || report 1 'action.yml wires require_proof input to VCR_REQUIRE_PROOF'
+
+# Path 1: the council scope lens.
+lens() { REQUIRE_PROOF="$1" node -e 'import("./scripts/council-config.mjs").then(m => process.stdout.write(m.LENSES.scope))'; }
+( cd "$ROOT" && lens false | grep -q 'judge the evidence' ) && report 1 'scope lens drops the proof clause when off' || report 0 'scope lens drops the proof clause when off'
+( cd "$ROOT" && lens true  | grep -q 'judge the evidence' ) && report 0 'scope lens keeps the proof clause when on'  || report 1 'scope lens keeps the proof clause when on'
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)